ruby/lib/transformer/syskit/exceptions.rb

```diff
@@ -243,10 +261,57 @@ def pretty_print(pp)
     end
 
     # Exception raised when a needed frame is not assigned
-    class MissingFrame < RuntimeError; end
+    class MissingFrame < BaseException; end
 
     # Exception raised when a producer requires itself to function
-    class RecursiveProducer < RuntimeError; end
+    class RecursiveProducer < BaseException; end
 
-end
+    # Exception raised when the transformation configuration contains errors
+    class InvalidConfiguration < BaseException; end
+
+    # Exception raised when a transformation requested in #transformation_chain
+    # cannot be found
+    class TransformationNotFound < RuntimeError
+        attr_reader :from
+        attr_reader :to
+
+        def initialize(from, to)
+            @from = from
+            @to = to
+        end
+    end
```

Comment:
There's a syskit part and a non-syskit part to this library. Don't move things from the non-sykit into the syskit part.